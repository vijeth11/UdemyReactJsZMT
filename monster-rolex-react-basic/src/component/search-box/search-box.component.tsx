import React, { ChangeEventHandler } from "react";
import './search-box.component.css';

interface ISearchBoxProps{
    placeholder?:string; 
    handleChange:ChangeEventHandler<HTMLInputElement>;
};

export const SearchInput = (props:ISearchBoxProps) => {
    return (
        <input 
        className = 'search' 
        type='search'
         placeholder={props.placeholder} 
         onChange={props.handleChange}/>
    );
}