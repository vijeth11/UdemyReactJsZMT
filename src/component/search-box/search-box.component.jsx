import React from "react";
import './search-box.component.css';

export const SearchInput = (props) => {
    return (
        <input 
        className = 'search' 
        type='search'
         placeholder={props.placeholder} 
         onChange={props.handleChange}/>
    );
}