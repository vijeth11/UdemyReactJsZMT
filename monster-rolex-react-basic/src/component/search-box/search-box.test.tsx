import { shallow } from "enzyme"
import { SearchInput } from "./search-box.component"
let mockPlaceHolder:string = "test";
let mockCallback = (event: any) => console.log(event)

it("initialization test",() => {
    expect(shallow(<SearchInput placeholder={mockPlaceHolder} handleChange = {mockCallback}/>)).toMatchSnapshot();
})