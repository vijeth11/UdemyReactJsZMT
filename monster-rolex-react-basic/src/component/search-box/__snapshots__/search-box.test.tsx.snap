// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`initialization test 1`] = `
<input
  className="search"
  onChange={[Function]}
  placeholder="test"
  type="search"
/>
`;
